openedx/core/djangoapps/user_authn/cookies.py

```diff
@@ -77,80 +124,73 @@ def set_logged_in_cookies(request, response, user):
         HttpResponse
 
     """
-    cookie_settings = standard_cookie_settings(request)
+    # Note: The user may not yet be set on the request object by this time,
+    # especially during third party authentication.  So use the user object
+    # that is passed in when needed.
+    if user.is_authenticated and not user.is_anonymous:
+
+        _set_deprecated_logged_in_cookie(response, request)
+        _set_deprecated_user_info_cookie(response, request, user)
+        _set_jwt_cookies(response, request, user)
+        CREATE_LOGON_COOKIE.send(sender=None, user=user, response=response)
```

Comment:
Nit: I'd prefer names like `CREATE_LOGON_COOKIE_SIGNAL`.

Reply:
hmm.. we should come up with best practices around our Django signals - especially if they will become a supported API.  I'll keep this as is for now - since it was an already existing signal.